lib/datadog/appsec/api_security/lru_cache.rb

```
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+require 'forwardable'
+
+module Datadog
+  module AppSec
+    module APISecurity
+      # An LRU (Least Recently Used) cache implementation that relies on the
+      # Ruby 1.9+ `Hash` implementation that guarantees insertion order.
+      #
+      # WARNING: This implementation is NOT thread-safe and should be used
+      #          in a single-threaded context.
+      class LRUCache
+        extend Forwardable
+
+        def_delegators :@store, :clear, :empty?
+
+        def initialize(max_size)
+          @max_size = max_size.to_i
```

This will silently convert non-integer strings to 0, have you considered `Integer(max_size)` instead? Or requiring the value to be of a numeric type already?

---

I will change it to be type check without conversion 👍🏼